### PR TITLE
fix gradle builds

### PR DESCRIPTION
**Pull request by lenborje**
I have fixed the gradle builds:

- Needed javafx.graphics as a dependency.
- Also needed javafx.base _with a classifier_ since otherwise only the empty **unclassified** javafx.base jar would be automatically pulled in.
- Gradle 4.8. couldn't run the application with JDK 11. Got some strange security error. Upgrade to 4.9 fixed the problem.
- Added automatic selection of the platform-specific classifier.
- As a final bonus: Added a .gitignore (of temporary build and edit files)

### CODE REVIEW

**Comment by johanvos** [comment]
Good progress.
However, the empty jars are still on the module-path (since they are in the classpath), and that gives an error for me on linux: 

java.lang.module.FindException: Unable to derive module descriptor for /home/johan/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-graphics/11-ea+19/bb4cb6cc1611d8f1ba5a9aa07c8af95ab4a99395/javafx-graphics-11-ea+19.jar
Caused by: java.lang.IllegalArgumentException: javafx.graphics.11.ea.19: Invalid module name: '11' is not a Java identifier

**Comment by lenborje** [comment]
In the pull request, I got it working using the older 11.0.0-SNAPSHOT builds. I get the same error as you when using the 11-ea+19 builds. The root problem is that gradle apparently also includes the non-platform-specifc jars (which are empty) on the class path, in addition to the specific jars. To work around this problem the empty jars should have the "Automatic-Module-Name" attribute set in the manifest, as suggested by Steve Hruda in [this mail to openjfx-dev](http://mail.openjdk.java.net/pipermail/openjfx-dev/2018-August/022285.html). But this would of-course also require a new EA build of the jars to be pushed to the Maven repository.

**Comment by lenborje** [comment]
I've updated my PR branch with a work-around in build.gradle (for hello3d and mesh) which makes it possible to run these two programs with gradle, JDK 11, and openjfx 11-ea+19.

**Comment by lenborje** [comment]
I've tried to upgrade to openjfx 11-ea+23, but then running **from gradle** fails with:

```> Task :run
RenderJob.run: internal exception
java.lang.ArrayIndexOutOfBoundsException: 0
        at java.base/java.util.Arrays$ArrayList.get(Arrays.java:4351)
        at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1306)
        at javafx.graphics/com.sun.glass.ui.Screen.getMainScreen(Screen.java:61)
        at javafx.graphics/com.sun.prism.sw.SWPipeline.getDefaultResourceFactory(SWPipeline.java:82)
        at javafx.graphics/com.sun.prism.GraphicsPipeline.getDefaultResourceFactory(GraphicsPipeline.java:120)
        at javafx.graphics/com.sun.javafx.tk.quantum.QuantumRenderer.lambda$createResourceFactory$2(QuantumRenderer.java:161)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at javafx.graphics/com.sun.javafx.tk.RenderJob.run(RenderJob.java:58)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at javafx.graphics/com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:125)
        at java.base/java.lang.Thread.run(Thread.java:831)
```

Running from maven works OK. I'm stumped...

File: topics/javafx3d/hello3d/build.gradle [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
+
+
 compileJava {
     doFirst {
-        println classpath.asPath
+//        println classpath.asPath
         options.compilerArgs = [
             '--module-path', classpath.asPath,
+            '--add-modules', 'javafx.controls',
+            '--add-modules', 'javafx.graphics'
```
Comment (johanvos) [inline topics/javafx3d/hello3d/build.gradle:25]: this should not be needed. Once we're in the Java layer, the modules are fine (i.e. controls requires graphics)

File: topics/javafx3d/hello3d/build.gradle [hunk 2]
```diff
@@ -1,31 +1,45 @@
...
     }
 }
 run {
     doFirst {
         jvmArgs = [
-            '--module-path', classpath.asPath,
-            '--add-modules', 'javafx.controls'
+            '--module-path', classpath.filter{!it.name.startsWith("javafx-") || it.name.contains("-$classifier")}.asPath,
+            '--add-modules', 'javafx.controls',
+            '--add-modules', 'javafx.graphics'
```
Comment (johanvos) [inline topics/javafx3d/hello3d/build.gradle:35]: same as above

**Review by johanvos — APPROVED** [review]

**Pull request merged**